waspc/cli/exe/Main.hs

```diff
@@ -98,12 +98,14 @@ main = withUtf8 . (`E.catch` handleInternalErrors) $ do
             projectName
             appDescription
             projectConfigJson
-      _ -> printWaspNewAiUsage
+      _unknownCommand -> printWaspNewAiUsage
     Command.Call.Start -> runCommand start
     Command.Call.StartDb -> runCommand Command.Start.Db.start
     Command.Call.Clean -> runCommand clean
     Command.Call.Compile -> runCommand compile
-    Command.Call.Db dbArgs -> dbCli dbArgs
+    Command.Call.Db dbArgs -> case dbArgs of
+      ["start"] -> runCommand Command.Start.Db.start
+      _commandOnDb -> runCommandOnRunningDb dbArgs
```

Comment:
Ok! What's a bit weird now is that if it is not a "normal" command then we assume it is command running on db, well thtat is ok, but then we also assume that `runCommandOnRunningDb` will print usage if no command is found. Which is weird a bit, that that logic got moved into `runCommandOnRunningDb`. Also, that logic will be printing info about `wasp db start`, which is not even one of the commands in the case inside that function. So this is looking weird.

One way to improve this is to inline it directly here. It will be a bit crowded though.

Another way is to go back to waht we had before and just add comments, maybe that is best.

Third way is to make "db commands that don't need db running" and "db commands that need db running" equivalent, something like this:

```hs
Command.Call.Db dbArgs -> dbCli dbArgs

dbCli :: [String] -> IO ()
dbCli args = fromMaybe printDbUsage $
  runCommand <$> parseRegularCmd args
  <|> runDbCommand <$> parseCmdThatRequiresRunningDb args
  where
    parseRegularCmd args' = case args' of
      ["start"] -> Just Command.Db.Reset.reset
      _ -> Nothing

    parseCmdThatRequiresRunningDb args' = case args' of
      ["reset"] -> Just Command.Db.Reset.reset
      "migrate-dev" : optionalMigrateArgs -> Just $ Command.Db.Migrate.migrateDev optionalMigrateArgs
      ["seed"] -> Just $ Command.Db.Seed.seed Nothing
      ["seed", seedName] -> Just $ Command.Db.Seed.seed $ Just seedName
      ["studio"] -> Just Command.Db.Studio.studio
      _ -> Nothing
```

I haven't tried compiling the code so maybe I wrote some nonsense but the idea is to have two parsers that return Maybe and connect them with Alternative (`<|>`) and if both fail go with `printDbUsage`.

Reply:
> ... but then we also assume that runCommandOnRunningDb will print usage if no command is found. Which is weird a bit, that that logic got moved into runCommandOnRunningDb. Also, that logic will be printing info about wasp db start

Ahh, I did consider that. I thought it was a little messy (having a catch-all in a nested function), but I didn't notice that it will also print info about `wasp db start`. Yeah, that's bad.

We can also do this:
```haskell
Command.Call.Db dbArgs -> case dbArgs of
  ["start"] -> runCommand Command.Start.Db.start
  _commandOnDb -> fromMaybe printDbUsage (runCommandOnRunningDbIfValid dbArgs)

runCommandOnRunningDbIfValid :: [String] -> Maybe (IO ())
runCommandOnRunningDbIfValid args = Command.Db.runCommandOnRunningDb <$> parsedCommand
  ...
```
That solves all the problems I think, but is too weird for my taste.

About the parser suggestion - also too complicated IMO. Thanks for sharing it though, the mechanics are interesting.

> Another way is to go back to what we had before and just add comments, maybe that is best.

Yes, that's my favorite option now. I was opposed to it at first because I really wanted the express constraints through code rather than through comments, but in this case the complexity isn't worth it.

@Martinsos Let me know if you by any chance prefer the `runCommandOnRunningDbIfValid` approach. If not, I'm sticking with the comment.

Reply:
Yeah, not worth it. What you presented now is better, but let's say I want to add a third class of db commands or something here, then this printUsage is again stuck under comandOnDb case, it is a bit weird, it is still not clear this is actually the third case next to normal command and db running command.
Which is why I liked a bit better the code I proposed because it is more explicit in the sense of "it is a normal command, or command on running db, or I will print usage", and it allows for an modification of that on the high level (like inserting a new case) or adding commands to each of the groups.

But yeah, if you think that is too complex, let's do the comments and we are good. Plus maybe longer name for runDbCommand, that is good change.